lms/djangoapps/course_api/api.py

```diff
@@ -0,0 +1,83 @@
+"""
+Course API
+"""
+
+from django.contrib.auth.models import User
+from django.template import defaultfilters
+from rest_framework import status
+from rest_framework.response import Response
+
+from opaque_keys.edx.keys import CourseKey
+
+from xmodule.course_module import DEFAULT_START_DATE
+
+from courseware.courses import (
+    get_courses,
+    course_image_url,
+    get_course_about_section,
+)
+
+
+def has_permission(request, username):
+
+    if not username:
+        return False
+
+    user = request.user
+    return user and (user.is_staff or user.username == username)
+
+
+def course_view(request, course_key_string):
+
+    course_key = CourseKey.from_string(course_key_string)
+    course_usage_key = modulestore().make_course_usage_key(course_key)
+
+    return Response({
+        'blocks_url': reverse(
+                    'blocks_in_block_tree',
+                    kwargs={'usage_key_string': unicode(course_usage_key)},
+                    request=request,
+                    )
+    })
```

Comment:
Generating a response seems out-of-scope for api.py. That seems like the view's responsibility.  api.py should be constructing and returning dicts (or better still, DRF Serializer objects).  I'll work on that refactor.